MinStack.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Time Complexity: O(1) for all push(), pop(), top(), getMin() functions
+ * Spack Complexity: O(n) 
```

Comment:
As we discussed yesterday in class , we talk about individual function's space complexity. The space used here is in constructor , we don't take constructor's complexity